helix-core/src/test/java/org/apache/helix/TestHelper.java

```diff
@@ -799,7 +799,12 @@ public static boolean verify(Verifier verifier, long timeout) throws Exception {
     long start = System.currentTimeMillis();
     do {
       boolean result = verifier.verify();
-      if (result || (System.currentTimeMillis() - start) > timeout) {
+      boolean isTimedout = (System.currentTimeMillis() - start) > timeout;
+      if (result || isTimedout) {
+        if (isTimedout && !result) {
+          LOG.error("verifier time out, consider try longer timeout, stack trace{}",
+              Arrays.asList(Thread.currentThread().getStackTrace()));
```

Comment:
The stack trace will be printed in one line with this code. I suggest we use `new throwable().printstacktrace()`

Reply:
The is intentional one line. `new throwable().printstacktrace()` would give you multiple lines. That is used for true exception logging. 

This is not really exception. Just a hint that verify() did not finish in time. We don't want to mix both case for easy reading the test output.

This Arrays.asList is used also in other places in our test.